ethcore/src/json_tests/chain.rs

```diff
@@ -114,6 +117,72 @@ pub fn json_chain_test<H: FnMut(&str, HookType)>(json_data: &[u8], start_stop_ho
 	failed
 }
 
+mod no_proof_engine {
+	use ethereum::ethash::{Seal};
+	use engines::{self, Engine, EthEngine};
+	use machine::EthereumMachine;
+	use header::{Header, BlockNumber, ExtendedHeader};
+	use error::{BlockError, Error};
+	use unexpected::OutOfBounds;
+	use std::collections::BTreeMap;
+	use std::sync::Arc;
+	use block::ExecutedBlock;
+
+	/// Run engine but skip some validation for 'NoProof' configuration of json tests.
+	#[derive(Clone)]
+	pub struct NoProofEthashEngine(pub Arc<EthEngine>);
+
+	impl Engine<EthereumMachine> for NoProofEthashEngine {
+		/// No difficulty handle, skip some test.
+		fn verify_block_basic(&self, header: &Header) -> Result<(), Error> {
+			// check the seal fields.
+			let _ = Seal::parse_seal(header.seal())?;
+
+			if header.gas_limit() > &0x7fffffffffffffffu64.into() {
+				return Err(From::from(BlockError::InvalidGasLimit(OutOfBounds { min: None, max: Some(0x7fffffffffffffffu64.into()), found: header.gas_limit().clone() })));
+			}
+
+			Ok(())
+		}
+
+		/// No nonce handle, skip test.
+		fn verify_block_unordered(&self, _header: &Header) -> Result<(), Error> {
```

Comment:
What's the difference of this and `VerifierType::CanonNoSeal`? It looks like here in addition we skipped `verify_block_unordered` transaction basics verification, which I worry would give us some true negatives of failing tests.

Reply:
Yes 'check_seal' false will probably do it, I'll check that on monday, on ice till then.